internal/expr/prepare.go

```diff
@@ -130,7 +130,26 @@ func prepareOutput(ti typeNameToInfo, p *outputPart) ([]fullName, []typeMember,
 		return nil
 	}
 
-	// Case 1: Generated columns e.g. "* AS (&P.*, &A.id)" or "&P.*".
+	// Case 1: Function column e.g. "max(10,20) AS &P.id".
+	if p.funcCol {
+		// Output expressions containing functions do not save the columns in
+		// the output expression. This is because the function itself may
+		// contain input expressions.
+		if numColumns != 0 {
+			return nil, nil, fmt.Errorf("internal error: func output expression has unexpected columns")
+		}
+		// Output expressions containing functions are limited to only a single
+		// type. To include more than one function in a query multiple output
+		// expressions can be used.
+		if numTypes != 1 {
+			return nil, nil, fmt.Errorf("internal error: func output expression has more than one type")
+		}
+		if p.targetTypes[0].name == "*" {
+			return nil, nil, fmt.Errorf("cannot use sql function with asterisk in output expression: %q", p.raw)
+		}
```

Comment:
Can we add the "good" return statement to the end of this block? I think it makes reading the code easier as you do not have to evaluate the rest of the if statements below. And same for case 4 for consistency while we are at it.

Reply:
Changed to do this is a separate `prepareFunc`.